src/message.rs

```diff
@@ -337,9 +347,7 @@ impl Message {
                     msg.viewtype = row.get("type")?;
                     msg.state = row.get("state")?;
                     let error: String = row.get("error")?;
-                    if !error.is_empty() {
-                        msg.error = Some(error);
-                    }
+                    msg.error = MessageError::from_str(&error).ok();
```

Comment:
What if database contains a value that is not in the enum, for example if you restore a backup from newer version, or some error is reworded in a later version? Enum is an overkill IMO, string was good enough.

Reply:
Also the error can be any string if we received an NDN. Of course we can add another enum type `ndn(String)` but I don't see why.

Reply:
@Hocuri @link2xt  I didn't yet discover the NDN usecase - can we specify all the possible errors? I'm not a fan of String errors if we can avoid those, as documentation and display to users can be implemented easier.

Maybe we can introduce a `unknown(String)` variant for database entries which are not yet supported by the used core version?

Reply:
It's in `message.rs`, see `handle_ndn` function (called by `handle_reports` in `mimeparser.rs`), it passes the error to `set_msg_failed` which writes it directly into the database.

Message errors are deliberately free-form, they may contain technical info extracted from messages, like headers, NDN report itself etc. If we want to translate them, it has to be done in the core, via stock strings.

It's not an error in the sense of rust `Error` which we want to match on, it's just something user-readable displayed in the technical info for failed messages.

Reply:
@link2xt Ok, I would still prefer to put the ndn errors in an own enum variant like "Other(String)" so matching on encryption errors would not need a string comparison. But if you object I'm fine to reverse the change with the enum and go back to a String.

Reply:
> Of course we can add another enum type `ndn(String)` but I don't see why.

If you have a real use case (and not just "it's more rusty") then I'm not generally against it. But it's probably better to leave the decision to @link2xt who has a better overview of the code than me anyway.

Reply:
> Ok, I would still prefer to put the ndn errors in an own enum variant like "Other(String)" so matching on encryption errors would not need a string comparison.

Do you need to match on encryption error for some reason? I think errors were not supposed to be machine-readable, you are not expected to use them other than checking if there is an error or not.

If you have some usecase for matching on the error, it is probably beter to introduce a new message state, which would then be possible to also use from C, which can't match on MessageError anyway.

Reply:
@link2xt Yes I had the situation where I had to match. If I delete the data directory of my bot, it forgets its gpg key and will recreate some. If a person then wants to write to the bot, it can't decrypt the message. That's the point where I need to match and known that this error can be recovered by sending a message to that person which restarts autocrypt.

Reply:
An easy fix is to make your bot reply something to every message, like "an error occured, please resend the message" if there is an error.

Reply:
@link2xt sure, I know now that there's not many different errors which could happen if I receive a message. I didn't know this when I wrote the bot/started to write `get_error`.
I'll reverse the enum changes and try to write a documentation which helps other people dealing with the same problems as I had when writing my bot.